docs/hugo/content/design/ADR-2023-05-Installing-Only-Selected-CRDs.md

```diff
@@ -0,0 +1,174 @@
+subsequent upgrades. This could be dealt with by using a 
+[pre-install hook](https://stackoverflow.com/questions/55503893/helm-patch-default-service-account). We would create and
+patch the `ConfigMap` in the pre-install hook.
+
+ConfigMap structure:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: managed-resources
+  namespace: azureserviceoperator-system
+data:
+  "all": ""  # magic key meaning "*" which isn't supported in keys
+  microsoft.resources: "*"
+  microsoft.storage: "StorageAccount,StorageAccountsBlobService"
+```
+
+The following `patch` commands could be run either directly by the user or by our pre-install hook:
+
+* Addition of new resources: `kubectl patch configmap managed-resources -n azureserviceoperator-system --patch '{"data":{"microsoft.storage":"*"}}'`
+* Removal of old resources: `kubectl patch configmap managed-resources -n azureserviceoperator-system --type=json -p='[{"op": "remove", "path": "/data/microsoft.resources"}]'`
+
+If nothing is specified as part of the upgrade, the set of supported resources will not change.
+
+If the `ConfigMap` doesn't exist when we go to upgrade, we can assume we're coming from an older ASO version that didn't
+support the `ConfigMap` and default to creating one that signals to install all resources.
+
+Pros:
+* Meets all the requirements.
+* Works for both Helm and raw-YAML situations. Helm as discussed above and raw YAML by having users
+
+Cons:
+* Complicated to implement.
+* Not the behavior that Helm users would necessarily expect for Helm values.
+
+Open questions:
+* Do we even need to support a Helm value for configuring this? Technically speaking the only reason we would need to is
+  to enable ASO as an Extension or Addon, as there everything must be done through Helm. For users of Helm itself we likely
+  can just tell them to run a `kubectl` command afterwards to create the `ConfigMap`. That does make the Helm install 
+  experience worse though.
+* We may be able to create the `ConfigMap` as empty given Helms use of
+  [3 way strategic merge](https://helm.sh/docs/faq/changes_since_helm2/#improved-upgrade-strategy-3-way-strategic-merge-patches)
+  and then patch it afterwards.
+* Is it OK that _providers_ would be patchable, but you would need to include all the resources when patching something 
+  like `microsoft.storage: "StorageAccount,StorageAccountsBlobService"`? We could just disallow individual resource selection
+  for now, although that might eventually become a problem for certain large providers (`Microsoft.Network` comes to mind).
+
+See also Helm [post rendering](https://helm.sh/docs/topics/advanced/#post-rendering), which we could also consider.
+pre-install hook (or post-install) seems easier though.
+
+#### Answer 2c: Using a CRD
+
+We investigated doing this originally for [Helm chart size limitations](../ADR-2023-02-Helm-Chart-Size-Limitations)
+
+Pros:
+* Slightly more structured than a `ConfigMap`.
```

Comment:
```suggestion
* Significantly more structured than a `ConfigMap`.
```

Reply:
Dropped the qualifier and said "More structured than a `ConfigMap`"